addons/payment_stripe/tests/test_stripe.py

```diff
@@ -35,16 +35,30 @@ def test_webhook_notification_confirms_transaction(self):
         """ Test the processing of a webhook notification. """
         tx = self.create_transaction('redirect')
         url = self._build_url(StripeController._webhook_url)
+        with patch(
+            'odoo.addons.payment_stripe.controllers.main.StripeController'
+            '._verify_notification_signature'
+        ):
+            self._make_json_request(url, data=self.notification_data)
+        self.assertEqual(tx.state, 'done')
+
+    @mute_logger('odoo.addons.payment_stripe.controllers.main')
+    def test_webhook_notification_verify_token(self):
+        """ Test the processing of a webhook notification. """
+        tx = self.create_transaction(flow='redirect', operation='validation', tokenize=True)
+        url = self._build_url(StripeController._webhook_url)
         with patch(
             'odoo.addons.payment_stripe.controllers.main.StripeController'
             '._verify_notification_signature'
         ), patch(
             'odoo.addons.payment_stripe.models.payment_acquirer.PaymentAcquirer'
             '._stripe_make_request',
-            return_value={'status': 'succeeded'},
+            return_value=self.setup_intent_response,
         ):
-            self._make_json_request(url, data=self.notification_data)
-        self.assertEqual(tx.state, 'done')
+            self._make_json_request(
+                url, data=dict(self.notification_data, type="setup_intent.succeeded")
```

Comment:
The expected payload is the same for `setup_intent.succeeded` and `payment_intent.succeeded` ?
I was expecting something quite different, but if only the type is expected to differ, then it's ok :eyes: